### ISSUE TITLE
How To Add Channels For Browser Windows & Separate Outputs to OBS?

### ISSUE THREAD

**Issue by theMuzzl3** [eiz/SynchronousAudioRouter#69]
I can't quite figure this out, exactly... and I wish I could split "system audio" into two entries...

Is there a way to add a separate channel to REAPER for Opera browser only, a separate one for Firefox, and another for chrome?

I want to send in audio signals from separate live streams, from different twitch users... and mix them together, with effects.  And, then I would like to send each channel as a separate output, into OBS (so that I can mute one source in OBS, and still hear it in REAPER).

Can it send out more than one output, to OBS?  I'd like to also send the microphone as another separate line out to OBS.

I am using ASIO4ALL driver, on Windows 10 64 bit home edition.
REAPER 64, OBS 64.

**Comment by TekSoup** [comment]
i dont think so, the browsers do not have a audio device setting, they just go by the default you set for your windows desktop. let alone asio support. Programs like VLC and such let you change in the settings. You want more advanced audio and video, switch to Vmix it is better than OBS, but it cost $$$

**Comment by theMuzzl3** [comment]
I made a bit of headway, with VBCable.

With VBCable, I can send audio from Opera to a REAPER channel
(no voice meeter open, in Windows settings: App volume and deviec preferences, Opera out to VB Cable, REAPER input set to VB Cable, in REAPER preferences: audio device set to WASAPI, input as VB Cable, output to Realtek Speakers, in exclusive mode).  All of the sample rates had to be the same.

I could use ReaStream to send channels directly to OBS.

However, I would then need to spend $5 on VB Cable A & B, to get 2 more virtual cables.

Enter SAR.  I added Opera in the applications tab of the device, and its working when the active channel is set to receive audio from VB Cable (both in windows and in reaper asio config & channel input).  

So, the next step was to not rely on VB Cable.  I set Opera to send to the SAR Music channel in windows app volume and device preferences, and then made sure all inputs/outputs are active on REAPER preferences, armed the track to receive input from the music channel, and wallah.  OBS also can have 4 channels + 2 mic/aux inputs, so now I can set up SAR recording channels to go out to OBS.

However, it is a bit buggy (seems like on windows' end), as the REAPER app disappeared.  I tried opening chrome, and it wouldn't show up either.  When I open Firefox, a bunch of firefox instances show up in windows apps volume window and it kills everything that Opera was feeding to REAPER.  I'll let Microsoft know that they need to look into making that window less buggy.

The only problem with Opera is that, while it uses less resources, will only show up as 1 device in windows and REAPER SAR settings.  Firefox will show up as 1 device per window, and I am not sure about Chrome.

**Comment by sunny8484** [comment]
hi, @theMuzzl3  you can use The application list to Enable per-application audio routing rules.
Applications displays the defined routing rules. Each entry specifies which default endpoints should be used for one or more applications. you can change the default endpoints
